src/Shared/TryParseMethodCache.cs

```diff
@@ -81,5 +180,110 @@ private static MethodInfo GetEnumTryParseMethod()
             Debug.Fail("static bool System.Enum.TryParse<TEnum>(string? value, out TEnum result) not found.");
             throw new Exception("static bool System.Enum.TryParse<TEnum>(string? value, out TEnum result) not found.");
         }
+
+        private static bool TryGetDateTimeTryPareMethod(Type type, out MethodInfo? methodInfo)
+        {
+            methodInfo = null;
+            if (type != typeof(DateTime) && type != typeof(DateOnly) &&
+                 type != typeof(DateTimeOffset) && type != typeof(TimeOnly))
+            {
+                return false;
+            }
+
+            var staticDateMethods = type.GetMethods(BindingFlags.Public | BindingFlags.Static)
```

Comment:
Since you know the exact parameters you're looking for, just call `GetMethod(string name, Type[] parameterTypes)` and then you don't need to loop.